src/libraries/System.Reflection.Emit/src/System/Reflection/Emit/TypeBuilderImpl.cs

```diff
@@ -183,22 +183,28 @@ private void ValidateAllAbstractMethodsAreImplemented()
 
                         if ((targetMethod == null || targetMethod.IsAbstract) && !FoundInInterfaceMapping(method))
                         {
-                            throw new TypeLoadException(SR.Format(SR.TypeLoad_MissingMethod, method, FullName));
+                            throw new TypeLoadException(SR.Format(SR.TypeLoad_MissingMethod, method.Name, FullName));
                         }
                     }
                 }
             }
         }
 
         [UnconditionalSuppressMessage("ReflectionAnalysis", "IL2065:DynamicallyAccessedMembers", Justification = "Methods are loaded from this TypeBuilder. The interface methods should be available at this point")]
+        [UnconditionalSuppressMessage("ReflectionAnalysis", "IL2075:DynamicallyAccessedMembers", Justification =
+            "Somehow #pragma warning disable IL2075 doesn't suppress anymore, related to https://github.com/dotnet/runtime/issues/96646")]
         [UnconditionalSuppressMessage("ReflectionAnalysis", "IL2085:DynamicallyAccessedMembers", Justification = "Methods are loaded from this TypeBuilder")]
         private void CheckInterfaces(Type[] _interfaces)
         {
             foreach (Type interfaceType in _interfaces)
             {
-#pragma warning disable IL2075 // Analyzer produces a different warning code than illink. The IL2065 suppression takes care of illink: https://github.com/dotnet/runtime/issues/96646
-                MethodInfo[] interfaceMethods = interfaceType.GetMethods(BindingFlags.Public | BindingFlags.NonPublic | BindingFlags.Instance | BindingFlags.Static);
-#pragma warning restore IL2075
+                Type ifaceType = interfaceType;
+                if (interfaceType.IsConstructedGenericType &&
+                    IsConstructedFromTypeBuilder(interfaceType.GetGenericTypeDefinition(), interfaceType.GetGenericArguments()))
+                {
+                    ifaceType = interfaceType.GetGenericTypeDefinition();
+                }
```

Comment:
Is this logic something that needs a comment. It isn't obvious to me why this is needed. Is this a mitigation with an uninstantiated vs instantiated Generic type?

Reply:
As discussed offline I am going to remove interface implementation validation as its getting too complex when explicit interface implementation and ***Builder types involved in generics